app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/BeaconViewModel.kt

```diff
@@ -17,15 +19,26 @@ class BeaconViewModel @Inject constructor(private val beaconRepository: BeaconRe
     ViewModel() {
 
   private var _uiState = MutableStateFlow(UIState())
-
   val uiState: StateFlow<UIState> = _uiState
 
   init {
     val sampleBeacon =
         Beacon(
```

Comment:
Is there a particular reason for creating this dummy beacon everytime ? Can't we just have a Beacon? initialized to null that we update once the actual value is available ?